omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/RoleDelegatingCrudResource.java

```diff
@@ -0,0 +1,253 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.resource.impl;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.BooleanProperty;
+import io.swagger.models.properties.RefProperty;
+import io.swagger.models.properties.StringProperty;
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.Role;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.RepHandler;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.Date;
+
+/**
+ * Subclass of {@link DelegatingCrudResource} with helper methods specific to
+ * {@link Role}
+ *
+ * @param <T>
```

Comment:
What is the value of creating this new class?

Reply:
it didnt sit down well with me to extend the `RoleResource1_8` class extending directly from `DelegatingCrudResource`

Reply:
But still, it does not mean that you should create a new class.

Reply:
are you implying i should use `DataDelegatingCrudResource` instead? because thats the other option i had in mind.

or better yet, any suggestions of how i should go about it?

Reply:
i realized `DataDelegatingCrudResource` doesn't work for this case and none of the other BaseDelegatingCrudResource implementation classes did so i went with `DelegatingCrudResource`